petab/lint.py

```diff
@@ -622,7 +622,7 @@ def observable_table_has_nontrivial_noise_formula(
     """
 
     return not observable_df[NOISE_FORMULA].apply(
-        lambda x: re.match(r'^[\w_\.]+$', x) is not None
+        lambda x: re.match(r'^[\w_\.]+$', str(x)) is not None
```

Comment:
To make it more interesting, `str(x)` might be a float in scientific notation :sweat_smile: 
I think it would be easier to handle numbers separately.